sklearn/tree/tree.py

```diff
@@ -121,6 +122,12 @@ def fit(self, X, y, sample_weight=None, check_input=True,
                 if X.indices.dtype != np.intc or X.indptr.dtype != np.intc:
                     raise ValueError("No support for np.int64 index based "
                                      "sparse matrices")
+            if X.max() == np.finfo(DTYPE).max or \
```

Comment:
I now get a bit better what your original patch was doing. Fair enough that this is a bit of an absurd case to handle with a warning, and I don't think it is even an issue. You were right in the first place.

Reply:
I will remove the warning then.